CMakeLists.txt

```diff
@@ -1,119 +1,207 @@
 
 # Set up build type
 if(NOT CMAKE_BUILD_TYPE)
-    set(CMAKE_BUILD_TYPE Release
-        CACHE STRING "Valid build types are: \
-        Release (default), None, Debug, NoExternalLibs, RelWithDebInfo, MinSizeRel." FORCE)
+	set(CMAKE_BUILD_TYPE Release
+		CACHE STRING "Valid build types are: \
+		Release (default), None, Debug, WithExternalLibs, RelWithDebInfo, MinSizeRel." FORCE)
 endif()
+### Defaults
+
 message(STATUS "Build type is ${CMAKE_BUILD_TYPE}")
 
 enable_testing(true)
 
 if(CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND NOT CMAKE_SYSTEM_NAME MATCHES "Windows")
-    add_definitions(-D _GLIBCXX_USE_CXX11_ABI=0) # Needed for CasADi compatibility
+	add_definitions(-D _GLIBCXX_USE_CXX11_ABI=0) # Needed for CasADi compatibility
 endif()
 
 if(CMAKE_CXX_COMPILER_ID MATCHES "GNU" AND CMAKE_SYSTEM_NAME MATCHES "Windows")
-    # MinGW, change to .lib such that mex recognizes it
-    set(CMAKE_STATIC_LIBRARY_SUFFIX ".lib")
+	# MinGW, change to .lib such that mex recognizes it
+	set(CMAKE_STATIC_LIBRARY_SUFFIX ".lib")
 endif()
 
 if(CMAKE_CXX_COMPILER_ID MATCHES "GNU" OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99 -Wall -fdiagnostics-show-option")
-    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -fdiagnostics-show-option")
+	set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -std=c99 -Wall -fdiagnostics-show-option")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11 -Wall -fdiagnostics-show-option")
 elseif(CMAKE_CXX_COMPILER_ID MATCHES "MSVC")
-    set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} /MT")
-    set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} /MT")
+	set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} /MT")
+	set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} /MT")
 endif()
 
-message(STATUS "Installation directory is ${CMAKE_INSTALL_PREFIX}")
-
 # Rpath handling, see https://cmake.org/Wiki/CMake_RPATH_handling#What_is_RPATH_.3F
 set(CMAKE_SKIP_BUILD_RPATH FALSE)
 set(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE)
 set(CMAKE_INSTALL_RPATH "$ENV{CASADIPATH}/casadi")
 
-if(NOT CMAKE_C_COMPILER_ID MATCHES "MSVC")
-    set(ACADOS_WITH_HPMPC ON CACHE BOOL "Add HPMPC solver")
+if(CMAKE_BUILD_TYPE MATCHES WithExternalLibs)
+	set(ACADOS_WITH_HPMPC ON CACHE BOOL "Add HPMPC solver")
+	set(ACADOS_WITH_QORE ON CACHE BOOL "Add QORE solver")
+	set(ACADOS_WITH_OOQP ON CACHE BOOL "Add OOQP solver")
+	set(ACADOS_WITH_QPOASES ON CACHE BOOL "Add qpOASES solver")
+	set(ACADOS_WITH_QPDUNES ON CACHE BOOL "Add qpDUNES solver")
+	set(ACADOS_WITH_OSQP ON CACHE BOOL "Add OSQP solver")
 endif()
 
-if(EXISTS ${PROJECT_SOURCE_DIR}/external/OOQP AND NOT CMAKE_BUILD_TYPE MATCHES NoExternalLibs AND NOT CMAKE_SYSTEM_NAME MATCHES "dSpace")
-    set(ACADOS_WITH_OOQP ON CACHE BOOL "Add OOQP solver")
+## External lib checks (
+
+if(ACADOS_WITH_HPMPC MATCHES ON AND CMAKE_C_COMPILER_ID MATCHES "MSVC")
+	set(ACADOS_WITH_HPMPC OFF CACHE BOOL "HPMPC solver" FORCE)
+	message(WARNING "HPMPC is not compatible with MSVC")
```

Comment:
maybe there should be the same-looking `WARNING` message for all?

Reply:
thanks, fixed